pwiz_tools/Skyline/Controls/Graphs/AllChromatogramsGraph.cs

```diff
@@ -58,11 +60,31 @@ public partial class AllChromatogramsGraph : FormEx
         public AllChromatogramsGraph()
         {
             InitializeComponent();
+
+            _windowCreatedEvent = new ManualResetEvent(false);
+            HandleCreated += Notification_HandleCreated;
+
             toolStrip1.Renderer = new CustomToolStripProfessionalRenderer();
             _stopwatch = new Stopwatch();
             _multiFileWindowWidth = Size.Width;
         }
 
+        private void Notification_HandleCreated(object sender, EventArgs e)
+        {
+            _windowCreatedEvent.Set();
+        }
+
+        public void RemoveAsync()
+        {
+            // Avoid closing the ACG during CreateHandle()
+            var closeThread = new Thread(() =>

```

Comment:
I wonder whether you should use "ActionUtil.RunAsync" here so that it gets an exception handler (and also because it's good to centralize all of our thread creating code).